src/common/plugin.ts

```diff
@@ -47,10 +55,49 @@ export abstract class Plugin implements IPlugin {
 
     const response = this.container.channelService.hasPermission(channelName, this.permission);
     if (!response) {
-      message.reply(`Please use this command in a \`${this.permission}\` channel`);
+      const rooms = Object.values(Constants.Channels[this.permission]);
+      const totalChannels = rooms.length;
+      rooms.splice(3);
+
+      let addonText = '';
+
+      try {
+        const id = rooms
+          .filter((room) => {
+            return this.container.guildService
+              .getChannel(room)
+              .permissionsFor(message.member || '')
+              ?.has('VIEW_CHANNEL');
+          })
+          .map((room) => this.container.guildService.getChannel(room).id);
+
+        if (this.permission.toString() === 'Private') {
+          addonText = ` This is primarily the class channels, and any channels we haven't defined.`;
+        } else if (id.length === 0) {
+          addonText = ' There are no permanent channels of this type.';
+        } else if (id.length === 1) {
+          addonText = ` <#${id[0]}> is the only channel with this type.`;
+        } else {
```

Comment:
Instead of a bunch of else ifs
just return after each if
```
if(chan is private){
    tell user can only be in class
}

if(id.length === 0){
    blah;
    return;
}
```

Reply:
are else if's bad?

Reply:
Not necessarily, but a lot of the time, they are very pretty, this code is very dense, and could use a little spacing. It makes it easier to follow too